src/hqdn3d.cpp

```diff
@@ -145,70 +171,84 @@ static const VSFrameRef *VS_CC hqdn3dGetFrame(
     Hqdn3dData * usrData = reinterpret_cast<Hqdn3dData *>(*instanceData);
 
     if (activationReason == arInitial) {
-        vsapi->requestFrameFilter(std::max(0, n - 1), usrData->clip, frameCtx);
+        // if we skip some frames, filter the gap anyway
+        if (n > usrData->last_frame + 1 &&
+            n - usrData->last_frame <= usrData->restartLap + 1 &&
+            usrData->last_frame >= 0) {
+
+            for (int i = usrData->last_frame + 1; i < n; i++)
+                vsapi->requestFrameFilter(i, usrData->clip, frameCtx);
+        // if processing out of sequence, filter several previous frames to minimize seeking problems
+        } else if (n != usrData->last_frame + 1) {
+            int sn = std::max(0, n - usrData->restartLap);
+
+            for (int i = sn + 1; i < n; i++)
```

Comment:
Always appending using blocks is preferred

Reply:
Sorry, I don't understand. What am I appending here?

Reply:
for (...) {
   // sth
}
compared to
for (...)
    // sth